microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/CommandRetrier.java

```diff
@@ -165,17 +203,31 @@ public Object execute() throws Exception {
     /**
      * Creates a new command for each retry since Hystrix commands can only be
      * executed once. Fallback method is not overridden here to ensure all
-     * retries are executed.
+     * retries are executed. If running in async mode, this method will execute
+     * on a different thread.
      *
      * @return Object returned by command.
      */
     private Object retryExecute() throws Exception {
+        // Config requires use of appropriate context class loader
+        if (contextClassLoader != null) {
+            Thread.currentThread().setContextClassLoader(contextClassLoader);
```

Comment:
No need because this is a special thread pool that we use for async. So when a new thread is started, the loader is set and then returned to the pool awaiting its next use.